Startup crash: PatternSyntaxException: Syntax error in regexp pattern near index 4:

## Steps to reproduce
- Launch firefox from upstream/master **on device**

### Expected behavior
App starts

### Actual behavior
app crashes

### Device information
* Fire TV device: Stick
* Latest affected Firefox version: up/master 202345441ebdd778066efd9dae92c66cf68eb9fc
* Earliest affected Firefox version: ?
* Turbo Mode: ?

Suspected regression: a-c upgrade in https://github.com/mozilla-mobile/firefox-tv/issues/2432

```
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.tv.firefox.debug/org.mozilla.tv.firefox.MainActivity}: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 4:
    (?<major>[0-9]+)\.(?<minor>[0-9]+)(?:\.(?<patch>[0-9]+))?(?<metadata>[^0-9].*)?
        ^
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2473)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2541)
        at android.app.ActivityThread.access$800(ActivityThread.java:160)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5597)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 4:
    (?<major>[0-9]+)\.(?<minor>[0-9]+)(?:\.(?<patch>[0-9]+))?(?<metadata>[^0-9].*)?
        ^
        at java.util.regex.Pattern.compileImpl(Native Method)
        at java.util.regex.Pattern.compile(Pattern.java:411)
        at java.util.regex.Pattern.<init>(Pattern.java:394)
        at java.util.regex.Pattern.compile(Pattern.java:381)
        at kotlin.text.Regex.<init>(Regex.kt:89)
        at mozilla.components.concept.engine.utils.EngineVersion$Companion.parse(EngineVersion.kt:71)
        at mozilla.components.browser.engine.system.SystemEngine.getVersion(SystemEngine.kt:77)
        at org.mozilla.tv.firefox.MainActivity.onCreate(MainActivity.kt:113)
        at android.app.Activity.performCreate(Activity.java:6010)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1122)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2426)
```

--- a/app/lint.xml
+++ b/app/lint.xml
@@ -34,8 +34,4 @@
     <!-- We don't want to fail the build, nor do we want to to pollute our informational output. -->
     <issue id="Typos" severity="informational"/>
 
-    <!-- This is an Android Components lint check. As long as we are not using the components logger (support-base) here
-         we do not need to treat this as an error. More:
-         https://github.com/mozilla-mobile/android-components/tree/master/components/support/base -->
-    <issue id="LogUsage" severity="informational" />
 </lint>

--- a/app/src/main/java/org/mozilla/tv/firefox/utils/IntentUtils.java
+++ b/app/src/main/java/org/mozilla/tv/firefox/utils/IntentUtils.java
@@ -14,6 +14,7 @@ import androidx.annotation.StringRes;
 import androidx.appcompat.app.AlertDialog;
 
 import mozilla.components.browser.session.Session;
+import mozilla.components.concept.engine.EngineSession;
 import mozilla.components.concept.engine.EngineView;
 import org.mozilla.tv.firefox.R;
 import org.mozilla.tv.firefox.ext.ContextKt;
@@ -95,7 +96,7 @@ public class IntentUtils {
         if (fallbackUrl != null) {
             // This is not Kotlin code :(
             final Session session = ContextKt.getWebRenderComponents(context).getSessionManager().getSelectedSessionOrThrow();
-            ContextKt.getWebRenderComponents(context).getSessionUseCases().getLoadUrl().invoke(fallbackUrl, session);
+            ContextKt.getWebRenderComponents(context).getSessionUseCases().getLoadUrl().invoke(fallbackUrl, session, EngineSession.LoadUrlFlags.Companion.none());
             return true;
         }
 

--- a/build.gradle
+++ b/build.gradle
@@ -9,7 +9,7 @@ buildscript {
     ext.architecture_components_version = '2.0.0'
     ext.kotlin_version = '1.3.31'
     ext.coroutines_version = '1.0.1'
-    ext.moz_components_version = '1.0.0'
+    ext.moz_components_version = '2.0.0'
     ext.androidx_work_version = '2.0.1'
 
     ext.robolectric_version = '4.0.2' // required for SDK 28.

